test/analytics/tests/analytics_from_the_server.ts

```diff
@@ -58,87 +56,120 @@ export default function ({ getService }: FtrProviderContext) {
         },
       ]);
     });
+    describe('after setting opt-in', () => {
+      let actions: Action[];
+      let context: Action['meta'];
 
-    it('after setting opt-in, it should extend the contexts and send the events', async () => {
-      await ebtServerHelper.setOptIn(true);
+      before(async () => {
+        await ebtServerHelper.setOptIn(true);
+        actions = await getActions();
+        context = actions[1].meta;
+      });
 
-      const actions = await getActions(3);
+      it('it should extend the contexts and send the events', async () => {
```

Comment:
nit: I think we can still split this test:

1. It calls `optIn`, `extendContext` and `reportEvents` in the right order.
2. Context includes `pid` (as injected by the `analytics_plugin_a`).